internal/characters/traveleranemo/asc.go

```diff
@@ -0,0 +1,44 @@
+package traveleranemo
+
+import (
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/event"
+	"github.com/genshinsim/gcsim/pkg/core/player"
+	"github.com/genshinsim/gcsim/pkg/enemy"
+)
+
+const a4ICDKey = "traveleranemo-a4-icd"
+
+func (c *char) a4() {
+	c.Core.Events.Subscribe(event.OnTargetDied, func(args ...interface{}) bool {
+		if _, ok := args[0].(*enemy.Enemy); !ok {
+			return false
+		}
+		atk := args[1].(*combat.AttackEvent)
+		if atk.Info.ActorIndex != c.Index {
+			return false
+		}
+		if atk.Info.AttackTag != combat.AttackTagElementalArt {
+			return false
+		}
+		if c.StatusIsActive(a4ICDKey) {
+			return false
+		}
+
+		c.AddStatus(a4ICDKey, 300, true)
+
+		for i := 0; i < 5; i = i + 1 {
+			c.QueueCharTask(func() {
```

Comment:
This isn't enough, because now you have hitlag affecting later ticks even though they shouldn't even be queued yet.